staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -283,23 +280,23 @@ func (flags *ApplyFlags) ToOptions(cmd *cobra.Command, baseName string, args []s
 		return nil, err
 	}
 
-	openAPISchema, _ := flags.Factory.OpenAPISchema()
+	openAPISchema, _ := f.OpenAPISchema()
 
 	validationDirective, err := cmdutil.GetValidationDirective(cmd)
 	if err != nil {
 		return nil, err
 	}
-	validator, err := flags.Factory.Validator(validationDirective, fieldValidationVerifier)
+	validator, err := f.Validator(validationDirective, fieldValidationVerifier)
 	if err != nil {
 		return nil, err
 	}
-	builder := flags.Factory.NewBuilder()
-	mapper, err := flags.Factory.ToRESTMapper()
+	builder := f.NewBuilder()
```

Comment:
Would it be better to use `resource.NewBuilder` function instead this

Reply:
`factory.NewBuilder()` is an interface whereas `resource.NewBuilder` is a concrete function, which mean if you initialize the factory as a `TestFactory` instead of a normal `Factory`, then `f.Builder()` can be implemented and behave differently.

That's why I think `factory.NewBuidler()` is better